cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/SubmitApi/Babbage/Transaction.hs

```diff
@@ -0,0 +1,172 @@
+import qualified Data.Aeson.Lens as Aeson
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Base16 as Base16
+import qualified Data.List as List
+import qualified Hedgehog.Extras as H
+import           Lens.Micro
+import           Testnet.SubmitApi
+
+hprop_transaction :: Property
+hprop_transaction = H.integrationRetryWorkspace 0 "submit-api-babbage-transaction" $ \tempAbsBasePath' -> do
+  H.note_ SYS.os
+  conf@Conf { tempAbsPath } <- H.noteShowM $ mkConf tempAbsBasePath'
+  let tempAbsPath' = unTmpAbsPath tempAbsPath
+  work <- H.createDirectoryIfMissing $ tempAbsPath' </> "work"
+
+  let
+    sbe = ShelleyBasedEraBabbage
+    era = toCardanoEra sbe
+    tempBaseAbsPath = makeTmpBaseAbsPath $ TmpAbsolutePath tempAbsPath'
+    options = cardanoDefaultTestnetOptions
+      { cardanoNodes = cardanoDefaultTestnetNodeOptions
+      , cardanoSlotLength = 0.1
+      , cardanoNodeEra = AnyCardanoEra era -- TODO: We should only support the latest era and the upcoming era
+      }
+
+  TestnetRuntime
+    { configurationFile
+    , testnetMagic
+    , poolNodes
+    , wallets
+    } <- cardanoTestnet options conf
+
+  poolNode1 <- H.headM poolNodes
+
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+
+  execConfig <- H.mkExecConfig tempBaseAbsPath poolSprocket1
+
+  void $ procSubmitApi
+    [ "--config", configurationFile
+    , "--testnet-magic", show @Int testnetMagic
+    , "--socket-path", "FILEPATH"
+    ]
+
+  txbodyFp <- H.note $ work </> "tx.body"
+  txbodySignedFp <- H.note $ work </> "tx.body.signed"
+  txbodySignedBinFp <- H.note $ work </> "tx.body.signed.bin"
+
+  void $ execCli' execConfig
+    [ "babbage", "query", "utxo"
+    , "--address", Text.unpack $ paymentKeyInfoAddr $ head wallets
+    , "--cardano-mode"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--out-file", work </> "utxo-1.json"
+    ]
+
+  utxo1Json <- H.leftFailM . H.readJsonFile $ work </> "utxo-1.json"
+  UTxO utxo1 <- H.noteShowM $ H.noteShowM $ decodeEraUTxO sbe utxo1Json
+  txin1 <- H.noteShow =<< H.headM (Map.keys utxo1)
+
+  void $ execCli' execConfig
+    [ "babbage", "transaction", "build"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--change-address", Text.unpack $ paymentKeyInfoAddr $ head wallets
+    , "--tx-in", Text.unpack $ renderTxIn txin1
+    , "--tx-out", Text.unpack (paymentKeyInfoAddr (head wallets)) <> "+" <> show @Int 5_000_001
+    , "--out-file", txbodyFp
+    ]
+
+  void $ execCli' execConfig
+    [ "babbage", "transaction", "sign"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx-body-file", txbodyFp
+    , "--signing-key-file", paymentSKey $ paymentKeyInfoPair $ wallets !! 0
+    , "--out-file", txbodySignedFp
+    ]
+
+  let submitApiConf = SubmitApiConf
+        { tempAbsPath = unTmpAbsPath tempAbsPath
+        , configPath = configurationFile
+        , epochSlots = 2
+        , sprocket = poolSprocket1
+        , port = 8090
```

Comment:
Consider randomising ports using [allocateRandomPorts](https://github.com/input-output-hk/hedgehog-extras/blob/1ec284b5e728e2beabda7512a5e76eae8b667ebb/src/Hedgehog/Extras/Stock/IO/Network/Socket.hs#L59) as used [here](https://github.com/james-iohk/cardano-node/commit/0542cd5d608e95d87b2c8b989e742b6fa1754cfc#diff-079ade366e18980b6232319946cc005904a8c865dadf467951da1d30f2069055R108) to avoid collision when running on our Macs (I've been toldby DevX that the MacOs runners have virtually no isolation)